src/pages/index.js

```diff
@@ -64,68 +77,177 @@ const addCardFormValidator = new FormValidator(validationSettings, addCardForm);
 
 function handleImageClick(data) {
   imagePreviewPopup.open(data);
 }
 
+function handleUpdateAvatar({ link }) {
+  avatarModal.renderLoading(true);
+  api
+    .changeProfilePic({ avatar: link })
+    .then((data) => {
+      avatarModal.close();
+      userInfo.setUserInfo(data.name, data.description, data.avatar);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarModal.renderLoading(false);
+    });
+}
+
 function createCard(data) {
-  return new Card(data, cardSelector, handleImageClick).getView();
+  const card = new Card(
+    data,
+    cardSelector,
+    handleImageClick,
+    (card) => {
+      confirmDelete.open();
+      confirmDelete.setSubmitAction(() => {
+        return api
+          .deleteCard(card.getId())
+          .then(() => {
+            card.handleDeleteCard();
+            confirmDelete.close();
+          })
+          .catch((err) => {
+            alert("No Card Available", err);
+          });
+      });
+    },
+    (card) => {
+      api
+        .changeLikeCardStatus(card.getId(), !card.isLiked())
+        .then(() => {
+          card.handleLikeIcon();
+        })
+        .catch((err) => {});
+    }
+  );
+
+  return card.getView();
 }
 
 function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.title, inputValues.description);
-  // use the close method instead of this line
-  // closePopup(profileEditModal);
-  editProfilePopup.close();
+  editProfilePopup.renderLoading(true);
+  api
+    .setUserInfo({ name: inputValues.title, about: inputValues.description })
+    .then((data) => {
+      userInfo.setUserInfo(
+        inputValues.title,
+        inputValues.description,
+        data.avatar
+      );
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      //update button text to say save
+      editProfilePopup.renderLoading(false);
+    });
 }
 
 profileEditButton.addEventListener("click", () => {
   const userData = userInfo.getUserInfo();
-  console.log(userData);
   profileTitleInput.value = userData.userName;
   profileDescriptionInput.value = userData.userDescription;
   profileFormValidator.resetValidation();
   editProfilePopup.open();
 });
 
+profileImageButton.addEventListener("click", () => {
+  avatarModal.open();
+});
+
 addNewCardButton.addEventListener("click", () => {
   addNewCardPopup.open();
 });
 
 const cardList = new Section(
   {
-    items: initialCards,
     renderer: (data) => {
       cardList.addItem(createCard(data));
     },
   },
   cardsConfig.containerSelector
 );
 
-cardList.renderItems(initialCards);
-
-function handleCardSubmit(data) {
-  // evt.preventDefault();
-  cardList.addItem(createCard(data));
-  cardTitleInput.value = "";
-  cardLinkInput.value = "";
-  addCardFormValidator.disableButton();
-  addNewCardPopup.close();
+api
+  .getCardList()
+  .then((data) => [cardList.renderItems(data)])
+  .catch((err) => {});
```

Comment:
`        .catch((err) => {});
`

Please, handle the error in any way

At least this way:

```js
.catch(console.error);

```